src/workerd/jsg/resource.h

```diff
@@ -653,6 +653,16 @@ struct WildcardPropertyCallbacks<
   }
 };
 
+template<typename TypeWrapper>
+decltype(std::declval<TypeWrapper&>().jsgGetMemoryName()) typeWrapperName(TypeWrapper &typeWrapper) {
+  return typeWrapper.jsgGetMemoryName();
```

Comment:
`jsgGetMemoryName` is the wrong thing to use. Likely what we'll want is to integrate this somehow into `JSG_RESOURCE_TYPE` using the `Type` parameter to set the name.